puan/logic/plog/__init__.py

```diff
@@ -624,297 +628,137 @@ def to_b64(self, str_decoding: str = 'utf8') -> str:
-        return self.__class__(
-            *map(
-                operator.methodcaller("reduce", fixed),
-                filter(
-                    maz.compose(
-                        functools.partial(
-                            operator.eq,
-                            0
-                        ),
-                        maz.pospartial(
-                            fixed.get,
-                            [(1, 0)]
-                        ),
-                        operator.attrgetter("id")
-                    ),
-                    self.propositions
-                )
-            ),
-            value=abs(self.value)-sum(
-                map(
-                    maz.compose(
-                        functools.partial(
-                            operator.le,
-                            1
-                        ),
-                        maz.pospartial(
-                            fixed.get,
-                            [(1, 0)]
-                        ),
-                        operator.attrgetter("id"),
-                    ),
-                    self.propositions
-                )
-            ),
-            id=self._id,
-        )
-
-    def flatten(self) -> typing.Iterable["Proposition"]:
-
-        """
-            Putting this proposition and all its sub propositions into a flat list of propositions.
-
-            Examples
-            --------
-                >>> AtLeast(AtLeast("x","y",value=1,id="C"),AtLeast("a","b",value=1,id="B"),value=1,id="A").flatten()
-                [AtLeast(id='A', equation='+B+C>=1'), AtLeast(id='B', equation='+a+b>=1'), AtLeast(id='C', equation='+x+y>=1'), Proposition(id='a'), Proposition(id='b'), Proposition(id='x'), Proposition(id='y')]
+        try:
+            _id, sign, props, bias, bounds = short
+        except Exception as e:
+            raise Exception(f"tried to convert short into AtLeast propositions but failed due to: {e}")
 
-            Returns
-            -------
-                out : typing.Iterable[Proposition]
-        """
-        return sorted(
-            set(
-                itertools.chain(
-                    [self],
-                    *map(
-                        operator.methodcaller("flatten"),
-                        self.propositions
-                    ),
-                )
-            )
+        return AtLeast(
+            value=-1*bias,
+            propositions=props,
+            variable=puan.variable(_id, bounds),
+            sign=sign,
         )
 
-    def specify(self) -> typing.Union["AtLeast", "AtMost"]:
-
-        """
-            Returns a more specific variant of this proposition.
+class AtMost(AtLeast):
 
-            Returns
-            -------
-                out : typing.Union[AtLeast, AtMost]
-        """
+    """
+        A AtMost proposition is an at least expression with negative coefficients and positive bias 
+        (e.g. -x-y-z >= -1). However, sub propositions are given as ordinary as well as the value.
```

Comment:
The example is a bit misleading here, since you talk about negative coefficients and positive bias and here the bias seem negative since the equation is not written on "standard" form. Perhaps :math:`-x-y-z+1 \ge 0` is better?